app/src/main/java/com/elavon/converge/TransactionService.java

```diff
@@ -81,7 +108,8 @@ public void createTransaction(Transaction transaction, String requestId, IPoyntT
         public void processTransaction(final Transaction transaction, final String requestId, final IPoyntTransactionServiceListener listener) throws RemoteException {
             Log.d(TAG, "processTransaction: " + requestId);
             Log.d(TAG, "Transaction:" + transaction.toString());
-            transactionManager.processTransaction(transaction, requestId, listener);
+            //transactionManager.processTransaction(transaction, requestId, listener);
+            fetchOrderParameters(transaction, requestId, listener);
```

Comment:
hmm you shouldn't call this unless there is an OrderId in the transaction references.

Reply:
I took this part of the code from the sample provided by Poynt.

Reply:
Will the order id always be automatically included in the transaction references when an Order is set on the Payment, or does it have to be set manually/separately? Do you access the order id in the transaction references using the hard coded string "OrderId", or is there a final String variable which can be used to reference it?

Reply:
yeah sorry the sample code is wrong.  In the Transaction References, every transactionReference has a TransactionReferenceType enum. So you can simply check for that instead of doing string comparison.

public enum TransactionReferenceType {
    POYNT_ORDER("P"),
    POYNT_STAY("S"),
    CUSTOM("C");
}

Reply:
updated in new commit. please re-review. thanks!